detectors/node/opentelemetry-resource-detector-container/src/detectors/ContainerDetector.ts

```diff
@@ -51,35 +56,17 @@ export class ContainerDetector implements Detector {
     }
   }
 
-  private async _getContainerIdV1() {
+  private async _getContainerIdV1(): Promise<string | undefined> {
     const rawData = await ContainerDetector.readFileAsync(
       this.DEFAULT_CGROUP_V1_PATH,
       this.UTF8_UNICODE
     );
     const splitData = rawData.trim().split('\n');
-    for (const line of splitData) {
-      const lastSlashIdx = line.lastIndexOf('/');
-      if (lastSlashIdx === -1) {
-        continue;
-      }
-      const lastSection = line.substring(lastSlashIdx + 1);
-      const colonIdx = lastSection.lastIndexOf(':');
-      if (colonIdx !== -1) {
-        // since containerd v1.5.0+, containerId is divided by the last colon when the cgroupDriver is systemd:
-        // https://github.com/containerd/containerd/blob/release/1.5/pkg/cri/server/helpers_linux.go#L64
-        return lastSection.substring(colonIdx + 1);
-      } else {
-        let startIdx = lastSection.lastIndexOf('-');
```

Comment:
was the search for `-` intended to remove the prefixes (`docker-` etc)?
can the new code now fail to remove such prefixes for something which is not in the predefined list of know prefixes?

Reply:
we are taking care of more prefixes :   
  readonly CRIO = 'crio-';
  readonly CRI_CONTAINERD = 'cri-containerd-';
  readonly DOCKER = 'docker-';

earlier code only took care of docker - this change handles additional cases.